[Bug] Failed to build BE UT

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.


### Version

master

### What's Wrong?

Some compilation errors occur when building BE UT.
```shell
ccache /devel/compiler/bin/g++ -DBE_TEST -DBOOST_DATE_TIME_NO_LIB -DBOOST_STACKTRACE_USE_BACKTRACE -DHAVE_INTTYPES_H -DHAVE_NETINET_IN_H -DUSE_HADOOP_HDFS -I/Programs/doris/be/src/apache-orc/c++/include -I/Programs/doris/be/ut_build_ASAN/src/apache-orc/c++/include -I/Programs/doris/be/ut_build_ASAN/src/clucene/src/shared -I/Programs/doris/be/src/clucene/src/core -I/Programs/doris/be/src/clucene/src/shared -I/Programs/doris/be/src/clucene/src/contribs-lib -I/Programs/doris/be/src -I/Programs/doris/be/test -I/devel/opt/java/include -I/devel/opt/java/include/linux -isystem /Programs/doris/be/../gensrc/build -isystem /Programs/doris/thirdparty/installed/include -isystem /Programs/doris/thirdparty/installed/gperftools/include -isystem /Programs/doris/thirdparty/installed/include/lz4 -isystem /Programs/doris/thirdparty/installed/include/zstd -O0 -fsanitize=address -DADDRESS_SANITIZER  -std=gnu++20   -D OS_LINUX -fno-access-control -g -Wall -Wextra -Werror -pthread -fstrict-aliasing -fno-omit-frame-pointer -Wnon-virtual-dtor -Wno-unused-parameter -Wno-sign-compare -fdiagnostics-color=always -Wno-nonnull -Wno-stringop-overread -Wno-stringop-overflow -D__STDC_FORMAT_MACROS -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX=1 -DBRPC_ENABLE_CPU_PROFILER -DS2_USE_GFLAGS -DS2_USE_GLOG -msse4.2 -DUSE_MEM_TRACKER -DUSE_UNWIND -fprofile-arcs -ftest-coverage -DGTEST_USE_OWN_TR1_TUPLE=0 -MD -MT test/CMakeFiles/doris_be_test.dir/runtime/load_stream_test.cpp.o -MF test/CMakeFiles/doris_be_test.dir/runtime/load_stream_test.cpp.o.d -o test/CMakeFiles/doris_be_test.dir/runtime/load_stream_test.cpp.o -c /Programs/doris/be/test/runtime/load_stream_test.cpp
/Programs/doris/be/test/runtime/load_stream_test.cpp: In member function 'virtual void doris::LoadStreamMgrTest::StreamService::open_stream_sink(google::protobuf::RpcController*, const doris::POpenStreamSinkRequest*, doris::POpenStreamSinkResponse*, google::protobuf::Closure*)':
/Programs/doris/be/test/runtime/load_stream_test.cpp:364:45: error: ignoring return value of 'doris::PStatus* doris::POpenStreamSinkResponse::release_status()', declared with attribute 'nodiscard' [-Werror=unused-result]
  364 |                     response->release_status();
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~^~
In file included from /Programs/doris/be/test/runtime/load_stream_test.cpp:23:
/Programs/doris/gensrc/build/gen_cpp/internal_service.pb.h:36446:26: note: declared here
36446 | inline ::doris::PStatus* POpenStreamSinkResponse::release_status() {
      |                          ^~~~~~~~~~~~~~~~~~~~~~~
/Programs/doris/be/test/runtime/load_stream_test.cpp:384:41: error: ignoring return value of 'doris::PStatus* doris::POpenStreamSinkResponse::release_status()', declared with attribute 'nodiscard' [-Werror=unused-result]
  384 |                 response->release_status();
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~^~
/Programs/doris/gensrc/build/gen_cpp/internal_service.pb.h:36446:26: note: declared here
36446 | inline ::doris::PStatus* POpenStreamSinkResponse::release_status() {
      |                          ^~~~~~~~~~~~~~~~~~~~~~~
/Programs/doris/be/test/runtime/load_stream_test.cpp:392:37: error: ignoring return value of 'doris::PStatus* doris::POpenStreamSinkResponse::release_status()', declared with attribute 'nodiscard' [-Werror=unused-result]
  392 |             response->release_status();
      |             ~~~~~~~~~~~~~~~~~~~~~~~~^~
/Programs/doris/gensrc/build/gen_cpp/internal_service.pb.h:36446:26: note: declared here
36446 | inline ::doris::PStatus* POpenStreamSinkResponse::release_status() {
      |                          ^~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

### What You Expected?

Build BE UT successfully.

### How to Reproduce?

```
./run-be-ut.sh
```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/be/test/runtime/load_stream_test.cpp
+++ b/be/test/runtime/load_stream_test.cpp
@@ -361,7 +361,7 @@ public:
                     cntl->SetFailed("Tablet not found");
                     status->set_status_code(TStatusCode::NOT_FOUND);
                     response->set_allocated_status(status.get());
-                    response->release_status();
+                    static_cast<void>(response->release_status());
                     return;
                 }
                 auto resp = response->add_tablet_schemas();
@@ -381,7 +381,7 @@ public:
                 cntl->SetFailed("Fail to accept stream");
                 status->set_status_code(TStatusCode::CANCELLED);
                 response->set_allocated_status(status.get());
-                response->release_status();
+                static_cast<void>(response->release_status());
                 return;
             }
 
@@ -389,7 +389,7 @@ public:
 
             status->set_status_code(TStatusCode::OK);
             response->set_allocated_status(status.get());
-            response->release_status();
+            static_cast<void>(response->release_status());
         }
 
     private:

